lrz_validator/main.cpp

```diff
@@ -0,0 +1,150 @@
+/*
+ Copyright 2024 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+#include <iostream>
+#include <vector>
+
+#include "dive_core/data_core.h"
+#include "pm4_info.h"
+
+int main(int argc, char **argv)
+{
+    Pm4InfoInit();
+    if ((argc != 2) && (argc != 3))
+    {
+        std::cout
+        << "You need to call: lrz_validator <input_file_name.rd> <output_file_name.txt>(optional)";
```

Comment:
Suggestion: We print to stdout whether LRZ test passed or not. And if the user wants to find out why it doesn't pass, then they can pass in an output_file_name.txt. In that case, maybe make the output_file truly optional, in that we don't output any files unless the user specifies it. Give them a bit more control, since in some cases their automated tests might just want to know whether the LRZ stuff passes or not, and don't care about the details?

Reply:
Their automated tests will probably parse stdout to see if the test passed or failed. Writing that into a verbose text file will force them to have to parse our complicated detailed output file, which could be a pain. That's why I suggest printout out a PASS/FAIL type of string in stdout only, with no details. So it's a predictable printout they can parse.

Reply:
Thanks for the suggestion, make sense. I made the change as suggested in the latest commit.